src/Microsoft.AspNet.Mvc.Core/ActionConstraints/ConsumesAttribute.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc.HeaderValueAbstractions;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Specifies the allowed content types which can be used to select the action based on request's content-type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class ConsumesAttribute : Attribute, IResourceFilter, IConsumesActionConstraint
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="ConsumesAttribute"/>.
+        /// </summary>
+        public ConsumesAttribute([NotNull] string contentType, params string[] otherContentTypes)
+        {
+            ContentTypes = GetContentTypes(contentType, otherContentTypes);
+        }
+
+        // The value used is a non default value so that it avoids getting mixed with other action constraints
+        // with default order.
+        /// <inheritdoc />
+        int IActionConstraint.Order { get; } = 200;
```

Comment:
don't have hidden internal values. Have the oder be public somewhere, so if someone wants to insert between these and default values he can. We typically used a value of -1000 elsewhere.